doc/tools/west/index.rst

```diff
@@ -8,278 +8,46 @@ named ``west`` (Zephyr is an English name for the Latin
 
-.. _west-history:
-
-History and motivation
+Command-Line Interface
 **********************
 
-West was added to the Zephyr project to fulfill two fundamental requirements:
-
-* The ability to work with multiple Git repositories
-* The ability to provide a user-friendly command-line interface to the Zephyr
-  build system and debug mechanisms
-
-During the development of west, a set of :ref:`west-design-constraints` were
-identified to avoid the common pitfalls of tools of this kind.
-
-Multiple Git Repositories
-=========================
-
-Zephyr intends to provide all required building blocks needed to deploy complex
-IoT applications. This in turn means that the Zephyr project is much more than
-an RTOS kernel, and is instead a collection of components that work together.
-In this context, there are a few reasons to work with multiple Git
-repositories in a standardized manner within the project:
-
-* Clean separation of Zephyr original code and imported projects and libraries
-* Avoidance of license incompatibilities between original and imported code
-* Reduction in size and scope of the core Zephyr codebase, with additional
-  repositories containing optional components instead of being imported
-  directly into the tree
-* Safety and security certifications
-* Enforcement of modularization of the components
-* Out-of-tree development based on subsets of the supported boards and SoCs
-
-See :ref:`west-multi-repo` for a detailed explanation of west's handling of
-multiple repository management.
-
-Command-line interface
-======================
-
-Zephyr uses CMake as its build system management tool, as described in
-:ref:`application`.
-While CMake is a very powerful system and language, its two-stage nature
-as a build system generator makes it non-trivial to present the user with
-a consistent set of command-line switches and options that enable advanced
-usage of the build tool ultimately responsible for building, flashing, and
-debugging the application.
-West was introduced to unify and streamline the developer's experience when
-working with Zephyr. It is very important to note that west does not
-replace the CMake-based build system, but directly uses it and augments it.
-
-See :ref:`west-flash-debug` for a detailed explanation of the flash and debug
-command-line interface exposed by west.
-
-.. _west-struct:
-
-Structure
-*********
-
-West structure
-==============
-
-West is downloaded and installed on a system in two stages:
-
-* Bootstrapper: Installed using ``pip``, implements ``west init``
-* Installation: Installed using ``west init``, implements all other commands
-
-Additional information about the two distinct parts of west can be found in
-the corresponding sections below.
-
-Repository structure
-====================
-
-Beyond west itself, the actual code repositories that west works with
-are the following:
+Like :program:`git` and :program:`docker`, the top-level :program:`west`
```

Comment:
The exception in Git is submodules, which takes options, sub-command and sub-sub-command (`git submodule update`)

Reply:
I.e. I don't think the fact that commands can take commands is really a problem. Though I *am* starting to wonder if we can rework our command API to make it easier for extension commands to not use argparse...

Reply:
Let's leave it as-is, I am fine with that.